src/AssetsPage/index.jsx

```diff
@@ -1,26 +1,37 @@
 import React from 'react';
+import statusMap from './statusMap.json';
 
 class AssetsPage extends React.Component {
-  static request() {
+  constructor(props) {
+    super(props);
+    this.state = {
+      apiConnectionStatus: 200,
+    };
+  }
+
+  request() {
     fetch('/api/assets/course-v1:edX+DemoX+Demo_Course/?page=0&page_size=50&sort=sort&asset_type=', {
       credentials: 'same-origin',
       headers: {
         Accept: 'application/json',
       },
     })
-      .then(data => data.json())
-      .catch(() => {
-        // console.log(error);
-      })
-      .then(() => {
-        // console.log(json);
+      .then((response) => {
+        if (response.status !== this.state.apiConnectionStatus) {
+          this.setState(() => ({ apiConnectionStatus: response.status }));
+        }
       });
   }
 
   render() {
-    AssetsPage.request();
+    this.request();
```

Comment:
This function might call `setState` which will trigger a re-render, causing this component to be rendered two times in a row initially if the API request fails. Maybe this call should go in the constructor. I don't think it needs to be called every render, just one time initially for now.